core/src/main/java/org/springframework/ldap/odm/core/impl/DefaultObjectDirectoryMapper.java

```diff
@@ -280,13 +282,23 @@ private void populateSingleValueAttribute(Object entry, LdapDataEntry context, F
         }
 
         // The Java representation of the LDAP entry
-        T result;
+        T result = null;
 
         ObjectMetaData metaData=getEntityData(clazz).metaData;
 
         try {
             // The result class must have a zero argument constructor
-            result = clazz.newInstance();
+            final Constructor<?>[] allConstructors = clazz.getConstructors();
```

Comment:
I think this could be simplified to be `Constructor<T> result = clazz.getDeclaredConstructor(parameterTypes);`